wfe/wfe.go

```diff
@@ -712,6 +715,102 @@ func (wfe *WebFrontEndImpl) UpdateAccount(
 	}
 }
 
+func (wfe *WebFrontEndImpl) KeyRollover(
+	ctx context.Context,
+	logEvent *requestEvent,
+	response http.ResponseWriter,
+	request *http.Request) {
+	// Extract and parse outer JWS, and retrieve account
+	body, outerHeaderURL, key, prob := wfe.verifyPOST(ctx, logEvent, request, wfe.lookupJWK)
+	if prob != nil {
+		wfe.sendError(prob, response)
+		return
+	}
+
+	existingAcct, prob := wfe.getAcctByKey(key)
+	if prob != nil {
+		wfe.sendError(prob, response)
+		return
+	}
+
+	// Extract inner JWS
+	parsedInnerJWS, err := wfe.parseJWS(string(body))
+	if err != nil {
+		wfe.sendError(acme.MalformedProblem(err.Error()), response)
+		return
+	}
+
+	innerHeaderURL, ok := parsedInnerJWS.Signatures[0].Header.ExtraHeaders[jose.HeaderKey("url")].(string)
+	if !ok || len(innerHeaderURL) == 0 {
+		wfe.sendError(acme.MalformedProblem("JWS header parameter 'url' required for inner JWS."), response)
+	}
+	if innerHeaderURL != outerHeaderURL {
+		wfe.sendError(acme.MalformedProblem("JWS header parameter 'url' differ for inner and outer JWS."), response)
+	}
+
+	newPubKey, prob := wfe.extractJWK(request, parsedInnerJWS)
+	if prob != nil {
+		wfe.sendError(prob, response)
+		return
+	}
+
+	// Copied from verifyJWS:
+	// TODO(@cpu): `checkAlgorithm()`
+
+	innerPayload, err := parsedInnerJWS.Verify(newPubKey)
+	if err != nil {
+		wfe.sendError(acme.MalformedProblem("JWS verification error"), response)
+		return
+	}
+
+	var innerContent struct {
+		Account string `json:"account"`
```

Comment:
You can get rid of the `json` tags on both of these fields since their name matches the JSON keys exactly and you aren't specifying any additional options in the tags.